### PR TITLE
[percep] Merge code into main

### PR DESCRIPTION
**Pull request by mmartin4972**
* Refactor for ZED SDK 3.0
* Implemented Point Cloud Obstacle Detection
* Cleaned and Restructured Code
* Added read from and write to file capabilities for Point Cloud, Depth, and RGB data

### CODE REVIEW

**Comment by mmartin4972** [comment]
Why does it say merge branch 'main' into main_merge2? I'm pretty sure I specified it to be the other way around

File: jetson/percep/pcl.hpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    PCL() : 
+        bearing{0}, distance{0}, detected{false},
+        pt_cloud_ptr{new pcl::PointCloud<pcl::PointXYZRGB>} {
+
+        #if ZED_SDK_PRESENT
+        sl::Resolution cloud_res = sl::Resolution(PT_CLOUD_WIDTH, PT_CLOUD_HEIGHT);
+        cloudArea = cloud_res.area();
+        #else
+        cloudArea = PT_CLOUD_WIDTH*PT_CLOUD_HEIGHT;
+        std::cerr << cloudArea<< endl;
```
Comment (raytitan) [inline jetson/percep/pcl.hpp:42]: No labels for this print?

File: jetson/percep/perception.hpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+#pragma GCC diagnostic pop
+
+#define THRESHOLD_NO_WAY  80000 //how will we calibrate if the rover width changes
+//#define THRESHOLD_NO_OBSTACLE_CENTER  80000
+//#define THRESHOLD_NO_SUBWINDOW 27000
+#define THRESHOLD_CONFIDENCE 90
+#define SKY_START_ROW 400
+#define BALL_DETECTION_MAX_DIST 6.0  // this number is obtained from experiment. if the distance of the detected ball is greater than this number, false detection, we should ignore
+#define BALL_DETECTION_MIN_RAD 5
+#define BALL_DETECTION_MAX_RAD 300
```
Comment (raytitan) [inline jetson/percep/perception.hpp:62]: What do you think of moving these into config files like nav and localization have

**Review by raytitan — CHANGES_REQUESTED** [review]
Do you need to commit the test images into the repository? We should probably only have code in here

File: jetson/percep/perception.hpp (hunk 2 above)
Comment (mmartin4972) [reply jetson/percep/perception.hpp:62]: We are currently working on that

File: jetson/percep/pcl.hpp (hunk 1 above)
Comment (mmartin4972) [reply jetson/percep/pcl.hpp]: Good catch

**Comment by mmartin4972** [comment]
> Do you need to commit the test images into the repository? We should probably only have code in here

I don't think we have any images in our repository? Maybe you are seeing deleted ones?

**Comment by mmartin4972** [comment]
Once I get everyone's approval I'll merge all commits after "[percep] Improved ground plane detection" into that commit

**Review by DavidSmith166 — APPROVED** [review]

**Review by amverni — DISMISSED** [review]
LGTM as long as you squash like you mentioned!

**Review by raytitan — APPROVED** [review]

**Review by DavidSmith166 — APPROVED** [review]

**Review by amverni — APPROVED** [review]

**Review by raytitan — APPROVED** [review]

**Pull request merged**